### PR TITLE
[1.2.0] [EASY] Refactor Test Skipper [API-1138]

### PR DESCRIPTION
**Pull request by yuce**
* Refactored the test skipper, so it can be tested better.
* Fixed the multiple condition behavior. All conditions must be satisfied to skip a test.
* More tests, documentation.
* Added `arch` key to `SkipIf`.

### CODE REVIEW

File: internal/it/skip.go [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+
+func mustAtoi(s string) int {
+	n, err := strconv.Atoi(s)
+	if err != nil {
+		panic(fmt.Errorf("could not parse int %s: %w", s, err))
+	}
+	return n
+}
+
+func checkEquality(left, operator, right, key string) bool {
```
Comment (utku-caglayan) [inline internal/it/skip.go:305]: Just a suggestion, case-insensitive comparison can be easier to use.

**Review by utku-caglayan — APPROVED** [review]
Looks great!

File: internal/it/skip.go (hunk 1 above)
Comment (yuce) [reply internal/it/skip.go:305]: We panic if a key (like `hz`, `version`) not found, so the user cannot make a mistake. I think let's keep it so the user cannot write something like `VeRsiON < 5.1` :)

**Review by emreyigit — APPROVED** [review]

**Comment by yuce** [comment]
@utku-caglayan @emreyigit Thanks for the review!

**Pull request merged**